UnitTests/OIDServiceDiscoveryTests.m

```diff
@@ -30,6 +30,20 @@
 #pragma GCC diagnostic push
 #pragma GCC diagnostic ignored "-Wgnu"
 
+// Define a subclass of @c OIDServiceDiscovery that provides the old implementation of
+// encodeWithCoder:.
+@interface OIDServiceDiscoveryOld : OIDServiceDiscovery
+
+@end
+
+@implementation OIDServiceDiscoveryOld
```

Comment:
Consider also testing the case where serialized service discovery objects are cached and shared between app groups.  Specifically: what happens if a non-updated member of an app group tries to read a cache containing the new serialization?

Reply:
An error was generated in the old decoding the new scenario.  I've now added forward compatibility and a test.